batch_finder_resource_method.md

```diff
@@ -0,0 +1,310 @@
+          } else if (currentCriteria.getId() == 2L) {
+            CollectionResult<Greeting, EmptyRecord> c2 = new CollectionResult<Greeting, EmptyRecord>(Arrays.asList(g2), 1);
+            batchFinderResult.putResult(currentCriteria, c2);
+          } else if (currentCriteria.getId() == 100L){
+            // on error: to construct error response for test
+            batchFinderResult.putError(currentCriteria, new RestLiServiceException(HttpStatus.S_404_NOT_FOUND, "Fail to find Greeting!"));
+          }
+        }
+    
+        return batchFinderResult;
+    }, _executor);
+
+  }
+```
+Every parameter of a BATCH_FINDER method must be annotated with one of:
+
+-   @`Context` - indicates that the parameter provides framework context
+    to the method. Currently all @`Context` parameters must be of type
+    `PagingContext`.
+-   @`QueryParam` - indicates that the value of the parameter is
+    obtained from a request query parameter. The value of the annotation
+    indicates the name of the query parameter. Duplicate names are not
+    allowed for the same BATCH_FINDER method.
+    For the batch parameter, the name must match the name in the method annotation.
+-   @`AssocKey` - indicates that the value of the parameter is a partial
+    association key, obtained from the request. The value of the
+    annotation indicates the name of the association key, which must
+    match the name of an @`Key` provided in the `assocKeys` field of the
+    @`RestLiAssociation` annotation.
+
+Parameters marked with @`QueryParam` and @`AssocKey`
+may also be annotated with @`Optional`, which indicates that the
+parameter is not required. *caution*: the batch parameter can not be optional.
+The @`Optional` annotation may specify a String value, indicating the default value to be used if the parameter
+is not provided in the request. If the method parameter is of primitive
+type, a default value must be specified in the @`Optional` annotation.
+
+Valid types for regular query parameters are:
+
+-   `String`
+-   `boolean` / `Boolean`
+-   `int` / `Integer`
+-   `long` / `Long`
+-   `float` / `Float`
+-   `double` / `Double`
+-   A Pegasus Enum (any enum defined in a `.pdsc` schema)
+-   Custom types (see the bottom of this section)
+-   Record template types (any subclass of `RecordTemplate` generated
+    from a `.pdsc` schema)
+-   Arrays of one of the types above, e.g. `String[]`, `long[]`, ...
+
+Valid type for batch criteria parameter:
+
+- Can only be Arrays of Record template type, if have to use some other data types like Pegasus Enum, etc as the array item,
+ need to wrap it into a Record Template (`.pdsc` schema)
+ 
+### Criteria Filter
+To implement a batch finder, the resource owner has to define a `RecordTemplate` to define a criteria filter parameter.
+The batch finder method will have to accept a array of this criteria filter.
+
+Example:
+
+The "GreetingCriteria" class represent a criteria filter to filter by "id" or by "tone".
```

Comment:
Can we swap and have the PDSC first, then the code generated from this PDSC?

Reply:
I suggested moving the PDSC example before the java class since the code is generated from the PDSC.